modular_skyrat/modules/morgue_radio/morgue.dm

```diff
@@ -0,0 +1,52 @@
+#define MORGUE_RADIO_COOLDOWN 1 MINUTES
+
+/obj/structure/bodycontainer/morgue/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/morgue_radio)
+
+/obj/structure/bodycontainer/morgue
+	/// reference to internal radio in the morgue trays.
+	var/obj/item/radio/headset/headset_med/radio
+
+/datum/component/morgue_radio
+	/// Typecasted reference to the current tray.
+	var/obj/structure/bodycontainer/morgue/morgue
+	/// Tracker for the morgue radio cooldown.
+	COOLDOWN_DECLARE(radio_cooldown)
+
+/datum/component/morgue_radio/Initialize(...)
+	. = ..()
+	morgue = parent
+	if(!istype(morgue))
+		return COMPONENT_INCOMPATIBLE
+
+/datum/component/morgue_radio/RegisterWithParent()
+	morgue.radio = new /obj/item/radio/headset/headset_med(morgue) // Initialize the radio in the morgue tray
```

Comment:
```suggestion
	// Initialize the radio in the morgue tray.
	// Put it in the connected tray because ironically that makes it not pop out when the morgue is opened.
	morgue.radio = new /obj/item/radio/headset/headset_med(morgue.connected)
```